LinearAlgebraInOSCAR.ipynb

```diff
@@ -0,0 +1,1003 @@
+   "id": "aa5a7942-0d7c-44f4-817e-d99dfdd8a428",
+   "metadata": {},
+   "source": [
+    "This tutorial provides an introduction to linear algebra in OSCAR. The aim is to describe important functionality and to exemplify it, so that users can start to use OSCAR more easily. Still, let us note that this tutorial does not aim to cover all of the existing functionality. Instead, we present a selection of arguably very important functions and features. For a more complete overview, please consult the documentation at https://docs.oscar-system.org/stable/LinearAlgebra/intro/.\n",
+    "\n",
+    "For this tutorial, the reader need not be familiar with OSCAR, but is assumed to have some basic understanding of Julia.\n",
+    "\n",
+    "The content of this notebook is as follows:\n",
+    "1. [Creating Matrices](#1-Creating-Matrices)\n",
+    "2. [Solving Linear Equation Systems](#2-Solving-Linear-Equation-Systems)\n",
+    "3. [Eigenvalues and Eigenspaces](#3-Eigenvalues-and-Eigenspaces)\n",
+    "4. [Normalforms](#4-Normalforms)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 69,
+   "id": "bd3322fa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "using Oscar\n",
+    "using Test"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "cddc1590",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "OSCAR version 1.1.1\n",
+      "  combining:\n",
+      "    AbstractAlgebra.jl   v0.41.9\n",
+      "    GAP.jl               v0.10.4\n",
+      "    Hecke.jl             v0.32.3\n",
+      "    Nemo.jl              v0.45.6\n",
+      "    Polymake.jl          v0.11.19\n",
+      "    Singular.jl          v0.23.4\n"
+     ]
+    }
+   ],
+   "source": [
+    "Oscar.versioninfo()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c1b3246b",
+   "metadata": {},
+   "source": [
+    "## 1 Creating Matrices"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fcb038f8-51b0-4b0d-8a6d-e88b815d411a",
+   "metadata": {},
+   "source": [
+    "There are many ways to create matrices in $\\texttt{OSCAR}$. We demonstrate a few:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "b734cc21-664d-4f40-b932-151cae872dd0",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[1   2   3]\n",
+       "[4   5   6]"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "m = matrix(ZZ, [[1,2,3],[4,5,6]])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "1306507b-f9d7-493f-a0da-4365b3ec47e4",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "ZZMatrix"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "typeof(m)"
```

Comment:
Is querying the type very useful for a tutorial at this point?

Perhaps more natural would be to create the matrix over ZZ. Then insert a sentence like: "We can create a matrix with the same coefficients over the rationals:"

And then after that: Note that these two matrices print the same but they have different types." (After which one can query the two types).

Then one could go on and show that one can compare and even add the two matrices anyway.

(And to pick up what you mentioned today, one could then introduce the finite field matrices and show that those can *not* be compared against the matrices in char 0. But all of that can also be done in a future refinement)